packages/edit-post/src/components/header/header-toolbar/index.js

```diff
@@ -64,9 +77,17 @@ function HeaderToolbar() {
 			),
 		};
 	}, [] );
+
 	const isLargeViewport = useViewportMatch( 'medium' );
 	const isWideViewport = useViewportMatch( 'wide' );
-
+	const { shouldShowContextualToolbar, canFocusHiddenToolbar } =
+		useShouldContextualToolbarShow( selectedBlockId );
+	// If there's a block toolbar to be focused, disable the focus shortcut for the document toolbar.
+	// There's a fixed block toolbar when the fixed toolbar option is enabled or when the browser width is less than the large viewport.
+	const blockToolbarCanBeFocused =
```

Comment:
Is there a similar thing going on in `edit-site` too? What is the advantage of using `useShouldContextualToolbarShow` here instead of leaving all the conditions inlined where they were and just looking at `( ( hasFixedToolbar || ! isLargeViewport ) && selectedBlockId )` here?

Reply:
There was a lot of [conditions in the block toolbar](https://github.com/WordPress/gutenberg/pull/49644/files#diff-d6d7994ce45fa28128923423e19291abd7de2fad164752f0f068cb9bfa93f4a0L97-L110) and @ajlende pointed out that we might be missing some conditions. For example, my conditions didn't account for smaller viewports. The way to be sure we weren't sending focus to the top toolbar was to use all the same conditions as we are for the block toolbar. That's why we decided to consolidate them. 

All that said, I don't have a strong preference. I do like this hook though. It can be very useful for knowing if the block toolbar should be showing or if it will be focusable from a keyboard shortcut anywhere else in Gutenberg.

Reply:
Having the hook means that we know it will be the same in both places which is what we want. From previous experience, if you don't do things this way in gutenberg, someone is going to update the condition in one place and miss the other place.